src/math/Quaternion.js

```diff
@@ -533,19 +533,21 @@ Object.assign( Quaternion.prototype, {
 
 		}
 
-		var sinHalfTheta = Math.sqrt( 1.0 - cosHalfTheta * cosHalfTheta );
+		var sqrSinHalfTheta = 1.0 - cosHalfTheta * cosHalfTheta;
 
-		if ( Math.abs( sinHalfTheta ) < 0.001 ) {
+		if ( sqrSinHalfTheta <= Number.EPSILON ) {
```

Comment:
`Math.abs()` is no longer needed?

Reply:
no : a square is always non negative !
do you prefer a update of this PR with a call to normalize() or a simply implementing slerp with a subcall to slerpFlat ?

Reply:
If `cosHalfTheta` = - 1.00001 due to roundoff then `sqrSinHalfTheta` could be negative.

Reply:
I agree, but that is no longer an issue : negative `sqrSinHalfTheta` corresponds to cases where the cos is round off errors away from 1 or -1, which corresponds to tiny values of the sin, for which lerp makes sense. (it WAS however an issue as the sqrt was not guarded against negative values and would output NaN values...)